src/mono/mono/mini/interp/interp.c

```diff
@@ -3499,15 +3531,11 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			return_offset = ip [1];
 			call_args_offset = ip [3];
 
-			if (csignature->hasthis) {
+			if (ftndesc->need_unbox) {
 				MonoObject *this_arg = LOCAL_VAR (call_args_offset, MonoObject*); 
-
-				if (m_class_is_valuetype (this_arg->vtable->klass)) {
-					gpointer unboxed = mono_object_unbox_internal (this_arg);
-					LOCAL_VAR (call_args_offset, gpointer) = unboxed;
-				}
+				LOCAL_VAR (call_args_offset, gpointer) = mono_object_unbox_internal (this_arg);
 			}
-			ip += 5;
+			ip += 4;
```

Comment:
@BrzVlad 

I expected to also see a change to something in `transform.c` to not push the `csignature` onto the instruction stream.  Is that happening in some helper somewhere?

Reply:
Yeah, I'm just overflowing some buffer with the signature index. Nothing to worry about ...